sstables/sstable_directory.cc

```diff
@@ -654,17 +654,11 @@ future<sstable_directory::pending_delete_result> sstable_directory::create_pendi
         try {
             auto trim_size = base_dir.native().size() + 1; // Account for the '/' delimiter
             for (const auto& sst : ssts) {
-                auto prefix = sst->_storage->prefix();
-                if (prefix.size() > trim_size) {
-                    out.write(prefix.begin() + trim_size, prefix.size() - trim_size).get();
-                    out.write("/").get();
-                }
-                auto toc = sst->component_basename(component_type::TOC);
-                out.write(toc).get();
+                auto toc = sst->toc_filename();
+                assert(toc.size() > trim_size);
```

Comment:
Why did you add this assert?

Reply:
Because it's going to trim `trim_size` bytes from `toc` front, so the toc string size() should be at least trim_size bytes. And it's not if+throw, because existing users make sure this is true -- the trim_size is `base_dir.size()` and `toc = base_dir + / + extra` for any toc that might pop-up here.

Reply:
We try to purge asserts from the codebase, or at least not add new ones. it is preferable to add if+throw (or just handle the false-case) if this is something you must check.

Reply:
OK, replaced assertion with internal error. There's no (easy?) way to handle the false-case here